compose/service.py

```diff
@@ -442,16 +442,20 @@ def _get_links(self, link_to_self):
 
     def _get_volumes_from(self):
         volumes_from = []
-        for volume_source in self.volumes_from:
-            if isinstance(volume_source, Service):
-                containers = volume_source.containers(stopped=True)
+        for volume_from in self.volumes_from:
+            if volume_from.mount_option:
+                mount_option = ':' + volume_from.mount_option
+            else:
+                mount_option = ''
+            if isinstance(volume_from.source, Service):
+                containers = volume_from.source.containers(stopped=True)
                 if not containers:
-                    volumes_from.append(volume_source.create_container().id)
+                    volumes_from.append(volume_from.source.create_container().id + mount_option)
                 else:
-                    volumes_from.extend(map(attrgetter('id'), containers))
+                    volumes_from.extend([container_id + mount_option for container_id in map(attrgetter('id'), containers)])
```

Comment:
Now that this is a comprehension, there's not much point in keeping the `map`+`attrgetter` stuff - it can just iterate over `containers`.